### ISSUE TITLE
[Bug] melonDS creates empty SRM-files which prevents saving the game

### ISSUE THREAD

**Issue by update-freak** [rommapp/romm#1080]
**RomM version**
v3.4.0

**Describe the bug**
I was not able to save the game (Pokemon Heartgold) with melonDS and also the firmware.bin selected. (The firmware.bin is necessary to avoid a communication error: https://github.com/Swordfish90/Lemuroid/issues/151)
In the screnshot I show the first start. After saving in the game and also use of the export save game a SRM-file was created but with only the entry "null".
![2024-08-11 09_25_24-RomM](https://github.com/user-attachments/assets/c51471f8-0b07-48c6-a5b6-8ece7705ad5c)

**To Reproduce**

1. Start Pokemon Heartgold / Soulsilver with melonDS and the firmware.bin selected
2. Save the game (with the ingame save function) and also use the export save button on the bottom of the screen -> then a SRM-file will be created
3. Close the game
4. Open the game again and try to load the save game

**Expected behavior**
The Savegame is saved and I did not have a SRM-file with the entry "null".

**Screenshots**
see above

**Desktop (please complete the following information):**

- OS: Windows 11
- Browser: Chrome 127

**Additional context**
I have also RetroArch on Android installed. When I use here the game with melonDS it is created a SAV-file instead of a SRM-file and there I can continue with the savegame. Maybe there is a difference with the melonDS core etc.?
Basically the idea is the have one save to play on RomM and RetroArch (Android), but RomM creates a SRM-file with "null" and RetroArch creates a SAV-file. Maybe RomM should here also create a SAV-file that both are compatible?

EDIT: I figured out that on RetroArch (Android) there is also an melonDS DS Core (https://www.reddit.com/r/RetroArch/comments/195cbnl/new_melonds_ds_core_released/) -> here the output of the savegame is also SRM, but RomM is not be able to use this SRM-file. It always starts a new game. Question is also if it's the same melonDS core as in RomM its only called melonDS and on RetroArch Android it's called melonDS DS.

**Comment by gantoine** [comment]
This is an underlying issue with emulatorjs and not something we can fix on our end.
